physical: ignore status message from aggregate stats collector

kev-cao opened the pull request:
#153893 taught PCR to update its status message based on the checkpoint events. However, as it also updates the status based on the range updates, this can cause race conditions that result in unexpected statuses. For now, we will ignore the status messages from the aggregate stats collector and keep the original behavior.

Epic: None

Fixes: #154230

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154253)
<!-- Reviewable:end -->

msbutler commented on pkg/crosscluster/physical/stream_ingestion_frontier_processor.go:
```diff
@@ -361,8 +361,16 @@ func (sf *streamIngestionFrontier) maybeUpdateProgress() error {
 		if replicatedTime.IsSet() && streamProgress.ReplicationStatus == jobspb.InitialScan {
 			streamProgress.ReplicationStatus = jobspb.Replicating
 			md.Progress.StatusMessage = streamProgress.ReplicationStatus.String()
-		} else if statusByStats != "" {
```
could we deflake this test by simply removing this condition added in the og PR? I feel like adding new metrics is a separate unit of work than updating the status message of the PCR. Your og PR didn't add any e2e testing for metrics, which is fine. But, i think modifying status updates deserves it's own testing.

how about quickly deflake the test by removing this condition, then in a follow up, we think about how incorporate the status message updates into pcr?

kev-cao committed:
physical: ignore status message from aggregate stats collector

#153893 taught PCR to update its status message based on the checkpoint
events. However, as it also updates the status based on the range
updates, this can cause race conditions that result in unexpected
statuses. For now, we will ignore the status messages from the aggregate
stats collector and keep the original behavior.

Epic: None

Fixes: #154230

msbutler commented:
bors r+ p=99
